From 09d6d4f84dd686e7370dc1d2292761cc11b55fe0 Mon Sep 17 00:00:00 2001
From: Jeffrey Ladouceur <jeffrey.ladouceur@freescale.com>
Date: Fri, 24 Aug 2012 00:38:36 +0000
Subject: [PATCH 04/45] DCE: add DCE device tree specs and binding specs

Add Decompressor Compressor device tree specs and binding specs.

Signed-off-by: Jeffrey Ladouceur <Jeffrey.Ladouceur@freescale.com>
(cherry picked from commit a9612ebb38699b21b92292fd8810a6ac3628f1d0)
Change-Id: I878150975dc372ea2d1f987b2468e9dddb9cb8e3
Reviewed-on: http://git.am.freescale.net:8181/1168
Reviewed-by: Schmitt Richard-B43082 <B43082@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Original patch taken from QorIQ-SDK-V1.8-20150619-yocto]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../devicetree/bindings/powerpc/fsl/fsl-dce.txt    |   66 ++++++++++++++++++++
 1 files changed, 66 insertions(+), 0 deletions(-)
 create mode 100644 Documentation/devicetree/bindings/powerpc/fsl/fsl-dce.txt

diff --git a/Documentation/devicetree/bindings/powerpc/fsl/fsl-dce.txt b/Documentation/devicetree/bindings/powerpc/fsl/fsl-dce.txt
new file mode 100644
index 0000000..7eb50d2
--- /dev/null
+++ b/Documentation/devicetree/bindings/powerpc/fsl/fsl-dce.txt
@@ -0,0 +1,66 @@
+=====================================================================
+DCE Device Tree Binding
+Copyright (C) 2012 Freescale Semiconductor Inc.
+
+ CONTENTS
+   -Overview
+   -DCE Node and Example
+
+NOTE: the DCE is also known as Freescale's Decompression and Compression Engine
+
+=====================================================================
+Overview
+
+DESCRIPTION
+
+    The Decompression and Compression Engine is an accelerator compatible
+    with Datapath Architecture providing lossless data decompression and
+    compression for the QorIQ family of SoCs.
+
+=====================================================================
+DCE Node
+
+Description
+
+    Node defines the base address of the DCE block.
+    This block specifies the address range of all global
+    configuration registers for the DCE block.  It
+    also receives error interrupts
+
+
+PROPERTIES
+
+   - compatible
+      Usage: required
+      Value type: <string>
+      Definition: Must include "fsl,dce"
+
+   - reg
+      Usage: required
+      Value type: <prop-encoded-array>
+      Definition: A standard property.  Specifies the physical
+          address and length of the DCE configuration registers.
+          registers
+
+   - interrupts
+      Usage: required
+      Value type: <prop_encoded-array>
+      Definition:  Specifies the interrupts generated by this
+           device.  The value of the interrupts property
+           consists of one interrupt specifier. The format
+           of the specifier is defined by the binding document
+           describing the node's interrupt parent.
+           This device only generates an error interrupt.
+
+   Note: All other standard properties (see the ePAPR) are allowed
+   but are optional.
+
+
+EXAMPLE
+	dce@312000 {
+		compatible = "fsl,dce";
+		reg = <0x312000 0x10000>;
+		interrupts = <16 2 1 4>;
+	};
+
+=====================================================================
-- 
1.7.5.4

